### ISSUE TITLE
Garbled Text Output When Rendering PDF 

### ISSUE THREAD

**Issue by constyn** [wojtekmaj/react-pdf#1684]
### Before you start - checklist

- [X] I followed instructions in documentation written for my React-PDF version
- [X] I have checked if this bug is not already reported
- [X] I have checked if an issue is not listed in [Known issues](https://github.com/wojtekmaj/react-pdf/wiki/Known-issues)
- [ ] If I have a problem with PDF rendering, I checked if my PDF renders properly in [PDF.js demo](https://mozilla.github.io/pdf.js/web/viewer.html)

### Description

When attempting to render a PDF, the text within the PDF is displayed as garbled and unreadable characters.

### Steps to reproduce

Load a PDF file through React-PDF
Observe the rendered output on the screen.

### Expected behavior

The PDF should be rendered with clear, readable text, displaying the content as intended.

### Actual behavior

The text in the rendered PDF is garbled with characters and symbols that do not represent the actual content of the document.

### Additional information

![image](https://github.com/wojtekmaj/react-pdf/assets/598240/59755abf-5c15-491d-afd7-0d0069b3ecd4)
- If the PDF is opened on the same system it is rendered ok
- It's occurring on the client's machine (which is a newer Dell model probably integrated graphics card)

### Environment

- **Browser Edge latest**:
- **OS Windows 11**:
- **React-PDF 7.6.0**:
- **React 18.2.0**:
- **vite 5.0.7**:

**Comment by wojtekmaj** [comment]
I don't think that's the same issue though. In #1010 you could see that the text React-PDF was _trying_ to render is more or less accurate, while OP's screenshot shows every single character being wrong.

I think client's graphics card is the key to reproducing this issue.

**Comment by IzabellaRegate** [comment]
[for my customers it is happening for either whole page or some texts]

I have small update on this topic. 
We did debugging session with affected customer and it seems issue is connected with text layer, grabled text is displayed properly while selecting it  🤔

**Comment by thordurhhh** [comment]
I am facing the same issue. There are a few users that have complained about this, I was sent a screenshot from a Chrome user with similar looking PDF to the one provided by @constyn 

But I am unable to reproduce the issue myself. Does anyone have a way of reproducing this?

React-PDF 7.5.0
React 18.2.0

**Comment by DonatienBdR** [comment]
We were facing the same issue with a PDF we generated ourselves (in a dedicated backend service). It also only happened for a few specific clients.
The solution/workaround we found is to embed the non-standard font we used when generating the PDF. We learned on the way that it's actually highly recommended to do so.
We also found [this comment](https://github.com/mozilla/pdf.js/issues/17597#issuecomment-1916648972) from the pdfjs library (among others) stating:
> Please note that the only way to guarantee that it works correctly is by embedding all fonts in the PDF document.

So, the root cause is likely related to the use of non-embedded custom fonts in the PDF, and the client's system fonts (which would explain why it is difficult to reproduce this issue).

I doubt react-pdf can do anything about this issue. Maybe pdfjs could, but it seems difficult nonetheless.

**Comment by sunilcygnet** [comment]
Hello Team,

I got the solution  this problem while using below code line.
 You can alternatively disable the textlayer prop in the <Page/> component by setting it to false although I wouldn't recommend that.

<Page pageNumber={1} renderTextLayer={false} /> 

Hope it might help many as they are suffering in production issue.
Regards &  empathy,
Sunil Rana

**Comment by TangMichael** [comment]
Facing the same issue here. This is new and only happens for windows users however not every user encounters this issue. Fonts used for the pdf is helvetica.

It's hard for to test because I cannot reproduce the issue. It makes me think it has to be something specific to the fonts on the user's computer possibly. 

React-pdf v7.1

Anyone has any ideas?

**Comment by thosil** [comment]
Upgrading to 9.0.0 fixed the issue, just struggled a little bit after the upgrade because the worker has to be imported with extension .mjs and not .js (first lines in the [upgrade guide](https://github.com/wojtekmaj/react-pdf/wiki/Upgrade-guide-from-version-8.x-to-9.x), but who reads the doc 😅)

**Comment by DaveEvans-Thalamos** [comment]
We've been experiencing this. Some clues I've found:
- This has been reported on pdf-js here: https://github.com/mozilla/pdf.js/issues/17109
- FireFox received a similar report, because they use pdf-js internally to render PDFs. The most enlightening comment is this one: https://bugzilla.mozilla.org/show_bug.cgi?id=1854090#c30
- The fix made by the Firefox team is only inside Firefox, so doesn't benefit users of react-pdf or pdf-js: https://phabricator.services.mozilla.com/D191639

TLDR: It seems like an Adobe product is making temporary fonts on Windows for the purpose of a print job, and they're not always getting cleaned up. Those incomplete fonts then get used when rendering your PDF on screen.

I'm still trying to find a mitigation which will work in all of our environments, and will report back here is and when I find one.

**Comment by DaveEvans-Thalamos** [comment]
I managed to reproduce this on Windows.

1. Download the attachment `Z@RD450.tmp` from this Firefox bug report : https://bugzilla.mozilla.org/show_bug.cgi?id=1854090
2. That file is actually a TTF font file. Rename it to have a `.ttf` extension to make drag and drop work in step 3.
3. In Windows, go to "Settings / Personalisation / Fonts". There should be a drag and drop target at the top to add a new font. Drag your tff file there.
4. Close and re-open your browser to make sure it picks up the new font.
5. Trying to render your PDF should now show the garbled characters.
6. Remember to unregister this font when you're done.

I found a workaround which works for me, but YMMV. (I only view PDFs which I generated, so I know exactly what they will contain). `pdf-js` has a [useSystemFonts](https://github.com/mozilla/pdf.js/blob/4c82438e61ed74148d0868184f7e6ebffe826bc5/src/display/api.js#L150) setting, which defaults to `true` in most browsers. Setting it explicitly to `false` bypasses the code path which would resolve in this broken font. 

You can pass this through in your `react-pdf` `Document` as follows:
```typescript
import { Document } from "react-pdf";

const MyComponent = () => {
  return <Document 
    options={{ useSystemFonts: false }}
    ...otherProps />;
}
```

**Comment by MartijnHols** [comment]
It seems `useSystemFonts` `false` is [not recommended](https://github.com/mozilla/pdf.js/issues/17675#issuecomment-1949928712) and I fear it will break more for my app which shows all kinds of user PDFs. I updated react-pdf to v9 and will check with users if this fixes the issue for them.

**Comment by jkgenser** [comment]
> It seems `useSystemFonts` `false` is [not recommended](https://github.com/mozilla/pdf.js/issues/17675#issuecomment-1949928712) and I fear it will break more for my app which shows all kinds of user PDFs. I updated react-pdf to v9 and will check with users if this fixes the issue for them.

Hi 👋 - I am the one that posted the link you referend. Yes, the maintainer said `useSystemFonts: false` is not recommended. However, it's the only way I have found to reliably prevent this issue where the incorrect font is loaded from the system.

I've been running `useSystemFonts: false` in production for a number of months now and it seems to work OK with a variety of PDFs. You need to also supply a standard font url.

**Comment by diogolessa** [comment]
> It seems `useSystemFonts` `false` is [not recommended](https://github.com/mozilla/pdf.js/issues/17675#issuecomment-1949928712) and I fear it will break more for my app which shows all kinds of user PDFs. I updated react-pdf to v9 and will check with users if this fixes the issue for them.

@MartijnHols any luck with v9?

**Comment by diogolessa** [comment]
> I've been running `useSystemFonts: false` in production for a number of months now and it seems to work OK with a variety of PDFs. You need to also supply a standard font url.

@jkgenser is setting `useSystemFonts: false` still work for you or have you found another way around? Thanks!
